packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1273,7 +1289,11 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         if (this.useSessionPool) {
             await this._timeoutAndRetry(
                 async () => {
-                    session = await this.sessionPool!.getSession();
+                    session = await this.sessionPool!.newSession({
```

Comment:
So does this mean that every request will get a fresh session? I'm not sure I understand the reasoning here

Reply:
Good point, my original reasoning was that this is what the `SessionPool` / `ProxyConfiguration` originally did anyway, but by making new `Session`s like this, we would probably run out of memory sooner or later. 

I'll revisit this :+1:

Reply:
I believe we discussed this in person - `SessionPool` is currently generating a large number of `Session`s before ever reusing one, so this behavior crudely approximates that for now.

I agree this is an intermediate step, and in `v4`, we should devise a better session reuse strategy.